File: 1006_clumsy_factorial/clumsy_factorial.java
class Solution {
    public int clumsy(int N) {
        Stack<Integer> stack = new Stack<Integer>();
        stack.push(N--);
        int position = 0;
        while (N > 0) {
            switch (position % 4) {
                case 0:
                    stack.push(stack.pop() * N);
                    break;
                case 1:
                    stack.push(stack.pop() / N);
                    break;
                case 2:
                    stack.push(N);
                    break;
                case 3:
                    stack.push(-N);
                    break;
            }
            N--;
            position++;
        }
        int ans = 0;
        while (!stack.isEmpty()) {
            ans += stack.pop();
        }
        return ans;
    }
}

File: 1006_clumsy_factorial/clumsy_factorial.py
import operator


class Solution:
    def clumsy(self, N: int) -> int:
        stack = [N]
        N -= 1
        poistion = 0
        while N > 0:
            if poistion % 4 == 0:
                stack.append(stack.pop() * N)
            elif poistion % 4 == 1:
                stack.append(int(operator.truediv(stack.pop(), N)))
            elif poistion % 4 == 2:
                stack.append(N)
            else:
                stack.append(-N)
            poistion += 1
            N -= 1
        return sum(stack)
